MekHQ/src/mekhq/gui/adapter/TOEMouseAdapter.java

```diff
@@ -280,22 +281,27 @@ public void actionPerformed(ActionEvent action) {
                             if (oldShip != null) {
                                 oldShip.unloadFromTransportShip(u);
                                 MekHQ.triggerEvent(new UnitChangedEvent(oldShip));
+                                gui.getTOETab().refreshForceView();
                             }
                         }
                     }
                     //now load the units
                     ship.loadTransportShip(units);
                     MekHQ.triggerEvent(new UnitChangedEvent(ship));
+                    gui.getTOETab().refreshForceView();
                 }
             }
         }
         if (command.contains(UNASSIGN_FROM_SHIP)) {
-            for (Unit u : units) {
-                for (UUID oldShipId : u.getTransportShipId().keySet()) {
-                    Unit oldShip = gui.getCampaign().getUnit(oldShipId);
-                    if (oldShip != null) {
-                        oldShip.unloadFromTransportShip(u);
-                        MekHQ.triggerEvent(new UnitChangedEvent(oldShip));
+            if (!units.isEmpty()) {
```

Comment:
This is superfluous with the for loop below. It'll do nothing if units is empty.